cabal.project

```diff
@@ -123,136 +123,151 @@ source-repository-package
   subdir: byron/crypto
 
 source-repository-package
   type: git
   location: https://github.com/input-output-hk/cardano-ledger-specs
-  tag: 9591f2d4b1915bcacd4797de563ca8a17d3b06f9
-  --sha256: 0xirm1rgawwdx2wj284hix0vfpm6w95rp315kmcwy5sd74qsynlc
+  tag: 0644881f33777a769103518705da6b86f2ff3441
+  --sha256: 0sj4qzgc829vrc5qcw8rs4dsd7fnbpc0wi0lpw1591m77hdf0lmz
   subdir: byron/crypto/test
 
 source-repository-package
   type: git
   location: https://github.com/input-output-hk/cardano-ledger-specs
-  tag: 9591f2d4b1915bcacd4797de563ca8a17d3b06f9
-  --sha256: 0xirm1rgawwdx2wj284hix0vfpm6w95rp315kmcwy5sd74qsynlc
+  tag: 0644881f33777a769103518705da6b86f2ff3441
+  --sha256: 0sj4qzgc829vrc5qcw8rs4dsd7fnbpc0wi0lpw1591m77hdf0lmz
   subdir: byron/chain/executable-spec
 
 source-repository-package
   type: git
   location: https://github.com/input-output-hk/cardano-ledger-specs
-  tag: 9591f2d4b1915bcacd4797de563ca8a17d3b06f9
-  --sha256: 0xirm1rgawwdx2wj284hix0vfpm6w95rp315kmcwy5sd74qsynlc
+  tag: 0644881f33777a769103518705da6b86f2ff3441
+  --sha256: 0sj4qzgc829vrc5qcw8rs4dsd7fnbpc0wi0lpw1591m77hdf0lmz
   subdir: shelley/chain-and-ledger/dependencies/non-integer
 
 source-repository-package
   type: git
   location: https://github.com/input-output-hk/cardano-ledger-specs
-  tag: 9591f2d4b1915bcacd4797de563ca8a17d3b06f9
-  --sha256: 0xirm1rgawwdx2wj284hix0vfpm6w95rp315kmcwy5sd74qsynlc
+  tag: 0644881f33777a769103518705da6b86f2ff3441
+  --sha256: 0sj4qzgc829vrc5qcw8rs4dsd7fnbpc0wi0lpw1591m77hdf0lmz
   subdir: shelley/chain-and-ledger/executable-spec
 
 source-repository-package
   type: git
   location: https://github.com/input-output-hk/cardano-ledger-specs
-  tag: 9591f2d4b1915bcacd4797de563ca8a17d3b06f9
-  --sha256: 0xirm1rgawwdx2wj284hix0vfpm6w95rp315kmcwy5sd74qsynlc
+  tag: 0644881f33777a769103518705da6b86f2ff3441
+  --sha256: 0sj4qzgc829vrc5qcw8rs4dsd7fnbpc0wi0lpw1591m77hdf0lmz
   subdir: shelley/chain-and-ledger/shelley-spec-ledger-test
 
 source-repository-package
   type: git
   location: https://github.com/input-output-hk/goblins
-  tag: 312198a1523736181ef7ddab15958bb32a9d9052
-  --sha256: 1w1b3g1s64rv4glrj416v1jmwnqhlz1zcqmz2y813jgs4d32m27k
+  tag: cde90a2b27f79187ca8310b6549331e59595e7ba
+  --sha256: 17c88rbva3iw82yg9srlxjv2ia5wjb9cyqw44hik565f5v9svnyg
 
 source-repository-package
   type: git
   location: https://github.com/input-output-hk/cardano-crypto/
   tag: 2547ad1e80aeabca2899951601079408becbc92c
   --sha256: 1p2kg2w02q5w1cvqzhfhqmxviy4xrzada3mmb096j2n6hfr20kri
+
+-- The r0 revision of this quickcheck-state-machine-0.7 on Hackage adds a lower
+-- bound on the text package: >=1.2.4.0. However, 1.2.4.0 doesn't support GHC
+-- 8.10, 1.2.4.1 will, but that hasn't been released yet. In the meantime,
+-- depend on the git version which is identical, but without the lower bound on
+-- text. See
+-- https://github.com/advancedtelematic/quickcheck-state-machine/issues/371
+source-repository-package
```

Comment:
Rather than vendoring q-s-m, we could instead just use `allow-older` to override the constraint on `text`:

```
allow-older: quickcheck-state-machine-0.7:text
```

Reply:
Good idea!